inc/shortcodes/class-pdf.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class PDF extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'PDF', 'shortcake-bakery' ),
+			'listItemImage'  => 'dashicons-media-document',
+			'attrs'          => array(
+				array(
+					'label'  => esc_html__( 'URL', 'shortcake-bakery' ),
+					'attr'   => 'url',
+					'type'   => 'text',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Turn embed code into a proper shortcode
+	 *
+	 * @param  string $content
+	 * @return string $content
+	 */
+	public static function reversal( $content ) {
+		$needle = '#http[^<]//[^<]+\.pdf?#';
```

Comment:
Sorry — I missed that you had added a reversal for this the first time.

Can you add a test to make sure linked PDFs don't get converted to embeds? e.g. it should always be on a newline, not already a shortcode, etc.

Also, for performance reasons, you should run `stripos( $content, 'pdf' )` before the regex to make sure there's a PDF in the body in the first place.